app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala

```diff
@@ -59,14 +77,27 @@ trait AssessmentCentreToFsbOrOfferProgressionService extends CurrentSchemeStatus
 
     val updates = FutureEx.traverseSerial(applications) { application =>
       FutureEx.futureToEither(application,
-        for {
-          currentSchemeStatus <- applicationRepo.getCurrentSchemeStatus(application.applicationId)
-          firstResidual = firstResidualPreference(currentSchemeStatus).get
-          _ <- maybeProgressToFsbOrJobOffer(application, firstResidual)
-        } yield ()
+        withErrLogging("Failed while progress to fsb or job offer") {
+          for {
+            currentSchemeStatus <- applicationRepo.getCurrentSchemeStatus(application.applicationId)
+            firstResidual = firstResidualPreference(currentSchemeStatus).get
+            _ <- maybeProgressToFsbOrJobOffer(application, firstResidual)
+          } yield ()
+        }
       )
     }
-
     updates.map(SerialUpdateResult.fromEither)
   }
-}
+
+  private def retrieveCandidateDetails(applicationId: String)(implicit hc: HeaderCarrier) = {
+    applicationRepo.find(applicationId).flatMap {
+      case Some(app) => authProviderClient.findByUserIds(Seq(app.userId))
```

Comment:
You can get these details from the application document, you don't need to add a dependency on the auth provider

Reply:
Actually I can't - there are no email there.

Reply:
candidate details collection

Reply:
ok, but that means I have to add dependency to contact-details repository, isn't it?

Reply:
True, but it means we don't have to call an external service